jwst/lib/reffile_utils.py

```diff
@@ -165,10 +169,15 @@ def ref_matches_sci(sci_model, ref_model):
     # Make sure the size attributes are consistent with data array
     if (xsize_sci != sci_model.shape[-1]) or \
        (ysize_sci != sci_model.shape[-2]):
-        log.warning("Science file data array size doesn't match subarray params")
-        log.warning("Using actual array size")
-        xsize_sci = sci_model.shape[-1]
-        ysize_sci = sci_model.shape[-2]
+
+        # Check for NIRSpec IRS2 mode, which is allowed to have a mismatch
+        if (ysize_sci == 2048) and (sci_model.shape[-2] == 3200):
+            ysize_sci = sci_model.shape[-2]
```

Comment:
What if xsize_sci didn't match sci_model.shape[-1]?  (This is in the section for a mismatch in either xsize_sci or ysize_sci.)  xsize_sci is from an xsize attribute, and it could have been set incorrectly, so it seems to me that xsize_sci should also be set (to sci_model.shape[-1]).

Reply:
Good point. Let me work on an update.